### ISSUE TITLE
After redacting the content, the position of the remaining text changes

### ISSUE THREAD

**Issue by hgdhot** [pymupdf/PyMuPDF#2248]
Hi,

I'm using 
```
import fitz

document = fitz.open("./11177948.pdf")
page = document.load_page(0)

redact_area = [
    fitz.Rect(311.2188415527344, 279.74359130859375, 554.0372314453125, 781.3309326171875),
    fitz.Rect(42.519996643066406, 756.7147216796875, 284.3482360839844, 789.4730834960938),
    fitz.Rect(42.51996612548828, 619.1275024414062, 285.2912292480469, 726.200439453125),
    fitz.Rect(130.93296813964844, 593.6478881835938, 197.8917236328125, 606.763671875),
    fitz.Rect(42.51995849609375, 304.8646240234375, 285.0877380371094, 575.4724731445312),
    fitz.Rect(144.4569549560547, 279.385009765625, 181.9504852294922, 292.3463439941406),
    fitz.Rect(126.3239517211914, 190.0526885986328, 468.9546813964844, 244.7281036376953),
    fitz.Rect(49.430450439453125, 156.88868713378906, 546.5343017578125, 181.19058227539062),
    fitz.Rect(66.8960189819336, 97.40320587158203, 529.7136840820312, 144.69845581054688)
]
for area in redact_area:
    page.add_redact_annot(area)

page.apply_redactions(images=1)
document.save('test.pdf')
```
to redact some content from the first page, after redaction, the position of the remaining text changes, Is this normal? how can I keep these remaining text in their original position?
Thanks a lot!
[example.pdf](https://github.com/pymupdf/PyMuPDF/files/10803438/example.pdf)
![a7235d600aeac195a14b7e0bfe1d9221](https://user-images.githubusercontent.com/19300626/220614417-3ddca155-ddbe-4c4e-bd24-4ae68bbab7f4.png)
![8abbab8cab64281a81929a663b6ab0a3](https://user-images.githubusercontent.com/19300626/220614780-9e7732a7-48df-4bde-aba5-ffb2f434571c.png)

**Comment by JorjMcKie** [comment]
I am underway currently, so just briefly a recommendation without having looked at details: execute page.clean_contents() immediately after loading the page.
Fat chance that this the reason. Please also have a look at the documentation under "misplaced insertions". This probably is no bug.

**Comment by JorjMcKie** [comment]
Oops sorry, answered the wrong issue.
Your issue is a duplicate, please see the issue list.

**Comment by hgdhot** [comment]
Is this a bug in MuPDF?

**Comment by hgdhot** [comment]
I also found that when performing redact operations, some text outside the redaction area will also be lost, which seems to be related to this problem？

**Comment by julian-smith-artifex-com** [comment]
I've tried your example, and it does not show the problem for me - the remaining text is in an unchanged position.

What PyMuPDF version are you using?

**Comment by hgdhot** [comment]
I'm using
Centos 6.3 64-bit
Python 3.7.6
PyMuPDF==1.21.1

>

**Comment by hgdhot** [comment]
Thank you for your reply，I'm using
Centos 6.3 64-bit
Python 3.7.6
PyMuPDF==1.21.1 
@julian-smith-artifex-com

**Comment by julian-smith-artifex-com** [comment]
Hmm, that's odd then. Just checking - how did you install PyMuPDF? Did you get it from pypi.org (e.g. with `python -m pip install pymupdf`) or build it yourself?

**Comment by hgdhot** [comment]
I installed PyMuPDF through `pip3 install PyMuPDF==1.21.1 `

**Comment by julian-smith-artifex-com** [comment]
Oh, apologies, i was mistakenly using an internal development version of PyMuPDF. With 1.21.1 i'm seeing the same problem as you.

Apparently there was a bug fix in MuPDF for this in January (so not included in PyMuPDF-1.21.1) which explains why i was not seeing it.

So, thanks for reporting this, and for the clear reproducer. It will be fixed in the next release.

**Comment by hgdhot** [comment]
Ok，thanks a lot! When the next version is planned to be released？
